src/TheThingsNetwork.cpp

```diff
@@ -237,8 +236,10 @@ bool TheThingsNetwork::personalize() {
 bool TheThingsNetwork::provision(const byte appEui[8], const byte appKey[16]) {
   sendCommand(F("mac set appeui"), appEui, 8);
   sendCommand(F("mac set appkey"), appKey, 16);
-  sendCommand(F("mac save"));
-  return waitForOK();
+  while (!sendCommand(F("mac save"))) {
+    delay(1000);
+  }
+  return waitForOK(10000);
```

Comment:
Did you test these new changes? I no longer get `ok` for `deveui` but for some reason still fails:

```
Sending: mac set appeui with 8 bytes
Sending: mac set appkey with 16 bytes
Sending: mac save
Wait for OK time-out expired
Sending: mac save
Sending: mac set deveui 0004A30B001B7AD2
Sending: mac join otaa
Join not accepted: denied
```